### DISCUSSION TITLE
Multiple Views of Output to Enhance Understanding of the Process

### DISCUSSION THREAD

**Discussion by K4KDR** [daniestevez/galileo-osnma#14] [General]
I wanted to share in case someone can use this method to better understand the timing of the various processes.

If the main command (shown here using a local hardware GNSS device feeding the Longan-nano via the Serial Client) is run in one terminal window with its output redirected to a TEXT file, that file can be monitored in other terminal windows with various degrees of filtering -and- with timestamps added.

In the example shown below, the following commands were used:

**Main command:** (from galileo-osnma/osnma-longan-nano-client/target/release)
`~/galmon/ubxtool --wait --port /dev/ttyACM0 --station 1 --stdout --galileo | ./osnma-longan-nano-client /dev/ttyUSB0 > /tmp/osnma-output.txt 2>&1`

**All messages:**
`tail -f /tmp/osnma-output.txt | ts '[%Y-%m-%d %H:%M:%S]'`

**Auth messages only:**
`tail -f -n0 /tmp/osnma-output.txt | grep --line-buffered -v NONE | grep --line-buffered AUTH | ts '[%Y-%m-%d %H:%M:%S]'`

![multi-term-view-with-time-stamps](https://user-images.githubusercontent.com/19886368/162120605-8ce0df3f-d3d9-45db-b7d9-612f02e8ae31.png)

**Comment by daniestevez** [comment]
I'm trying to clean up the issues by closing what is already solved and moving to a discussion what has background information that could be interesting to others. I think this is best moved to a discussion.
